pkg/asset/installconfig/powervs/client.go

```diff
@@ -268,6 +270,32 @@ func (c *Client) GetInstanceCRNByName(ctx context.Context, name string, publish
 	return "", fmt.Errorf("DNS zone %q not found", name)
 }
 
+// GetDNSCustomResolverIP gets the DNS Server IP of a custom resolver associated with the specified VPC subnet in the specified DNS zone.
+func (c *Client) GetDNSCustomResolverIP(ctx context.Context, dnsID string, vpcID string) (string, error) {
+
+	listCustomResolversOptions := c.dnsServicesAPI.NewListCustomResolversOptions(dnsID)
+	customResolvers, _, err := c.dnsServicesAPI.ListCustomResolversWithContext(ctx, listCustomResolversOptions)
+	if err != nil {
+		return "", err
+	}
+
+	subnets, err := c.GetVPCSubnets(ctx, vpcID)
+	if err != nil {
+		return "", err
+	}
+
+	for _, customResolver := range customResolvers.CustomResolvers {
+		for _, location := range customResolver.Locations {
+			for _, subnet := range subnets {
+				if *subnet.CRN == *location.SubnetCrn {
+					return *location.DnsServerIp, nil
+				}
+			}
+		}
+	}
+	return "", nil
```

Comment:
Maybe it's better to return an error here? When using this function you seem to assume if `err == nil` than the IP returned is valid.
```suggestion
	return "", fmt.Errorf("DNS server IP of custom resolver for %q not found", dnsID)
```

Reply:
yep good point, committed!